IKVMReference on Linux

Hello, I'm using the IKVM 8.2.0 nuget and seeing an issue when trying to build a project that uses the IKVMReference tag on Linux. I've been able to reproduce on the following hosts:

**Debian 11**
```
dylan@debian11:~$ uname -a
Linux debian11 5.10.0-10-amd64 #1 SMP Debian 5.10.84-1 (2021-12-08) x86_64 GNU/Linux
dylan@debian11:~$ dotnet --version
6.0.101
```


**Amazon Linux 2**
```
[ec2-user@amzn2-dylan ~]$ uname -a
Linux amzn2-dylan 4.14.281-212.502.amzn2.x86_64 #1 SMP Thu May 26 09:52:17 UTC 2022 x86_64 x86_64 x86_64 GNU/Linux
[ec2-user@amzn2-dylan ~]$ dotnet --version
6.0.301
```

<hr/>

When I go to build I'm seeing this:

```
dylan@debian11:~/Downloads/IKVM-TikadotNet-Harness/Tikadot6$ dotnet build
Microsoft (R) Build Engine version 17.0.0+c9eb9dd64 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
  Restored /home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj (in 433 ms).
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018: The "IkvmCompiler" task failed unexpectedly. [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018: IKVM.Tool.IkvmToolException: Could not set user executable bit on '/home/dylan/.nuget/packages/ikvm/8.2.0/bin/ikvmc/netcoreapp3.1/linux-x64/ikvmc'. [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018:  ---> System.DllNotFoundException: Unable to load shared library 'Mono.Unix' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libMono.Unix: cannot open shared object file: No such file or directory [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018:    at Mono.Unix.Native.Syscall.lstat(String file_name, Stat& buf) [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018:    at Mono.Unix.UnixFileSystemInfo.TryGetFileSystemEntry(String path, UnixFileSystemInfo& entry) [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018:    at Mono.Unix.UnixFileSystemInfo.GetFileSystemEntry(String path) [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018:    at IKVM.Tool.Compiler.IkvmCompilerLauncher.ExecuteAsync(IkvmCompilerOptions options, CancellationToken cancellationToken) in D:\a\ikvm\ikvm\src\IKVM.Tool\Compiler\IkvmCompilerLauncher.cs:line 394 [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018:    --- End of inner exception stack trace --- [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018:    at IKVM.Tool.Compiler.IkvmCompilerLauncher.ExecuteAsync(IkvmCompilerOptions options, CancellationToken cancellationToken) in D:\a\ikvm\ikvm\src\IKVM.Tool\Compiler\IkvmCompilerLauncher.cs:line 409 [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018:    at IKVM.MSBuild.Tasks.IkvmCompiler.Execute() in D:\a\ikvm\ikvm\src\IKVM.MSBuild.Tasks\IkvmCompiler.cs:line 297 [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]

Build succeeded.

/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018: The "IkvmCompiler" task failed unexpectedly. [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018: IKVM.Tool.IkvmToolException: Could not set user executable bit on '/home/dylan/.nuget/packages/ikvm/8.2.0/bin/ikvmc/netcoreapp3.1/linux-x64/ikvmc'. [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018:  ---> System.DllNotFoundException: Unable to load shared library 'Mono.Unix' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libMono.Unix: cannot open shared object file: No such file or directory [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018:    at Mono.Unix.Native.Syscall.lstat(String file_name, Stat& buf) [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018:    at Mono.Unix.UnixFileSystemInfo.TryGetFileSystemEntry(String path, UnixFileSystemInfo& entry) [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018:    at Mono.Unix.UnixFileSystemInfo.GetFileSystemEntry(String path) [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018:    at IKVM.Tool.Compiler.IkvmCompilerLauncher.ExecuteAsync(IkvmCompilerOptions options, CancellationToken cancellationToken) in D:\a\ikvm\ikvm\src\IKVM.Tool\Compiler\IkvmCompilerLauncher.cs:line 394 [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018:    --- End of inner exception stack trace --- [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018:    at IKVM.Tool.Compiler.IkvmCompilerLauncher.ExecuteAsync(IkvmCompilerOptions options, CancellationToken cancellationToken) in D:\a\ikvm\ikvm\src\IKVM.Tool\Compiler\IkvmCompilerLauncher.cs:line 409 [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018:    at IKVM.MSBuild.Tasks.IkvmCompiler.Execute() in D:\a\ikvm\ikvm\src\IKVM.MSBuild.Tasks\IkvmCompiler.cs:line 297 [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
/home/dylan/.nuget/packages/ikvm/8.2.0/buildTransitive/netstandard2.0/IKVM.targets(100,9): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/home/dylan/Downloads/IKVM-TikadotNet-Harness/Tikadot6/Tikadot6.csproj]
    0 Warning(s)
    1 Error(s)

Time Elapsed 00:00:01.71
MSBUILD : error MSB1025: An internal failure occurred while running MSBuild.
Microsoft.Build.Shared.InternalErrorException: MSB0001: Internal MSBuild Error: Request 5 expected to be in state Executing but state is actually Yielding
   at Microsoft.Build.Shared.ErrorUtilities.ThrowInternalError(String message, Object[] args)
   at Microsoft.Build.Shared.ErrorUtilities.VerifyThrow(Boolean condition, String unformattedMessage, Object arg0, Object arg1, Object arg2)
   at Microsoft.Build.BackEnd.SchedulableRequest.VerifyState(SchedulableRequestState requiredState)
   at Microsoft.Build.BackEnd.SchedulingData.ExpectScheduledRequestState(Int32 globalRequestId, SchedulableRequestState state)
   at Microsoft.Build.BackEnd.Scheduler.ReportResult(Int32 nodeId, BuildResult result)
   at Microsoft.Build.Execution.BuildManager.HandleResult(Int32 node, BuildResult result)
   at Microsoft.Build.Execution.BuildManager.ProcessPacket(Int32 node, INodePacket packet)
   at Microsoft.Build.Execution.BuildManager.<>c__DisplayClass76_0.<Microsoft.Build.BackEnd.INodePacketHandler.PacketReceived>b__0()
   at Microsoft.Build.Execution.BuildManager.ProcessWorkQueue(Action action)
--- End of stack trace from previous location ---
   at Microsoft.Build.Execution.BuildManager.EndBuild()
   at Microsoft.Build.CommandLine.MSBuildApp.BuildProject(String projectFile, String[] targets, String toolsVersion, Dictionary`2 globalProperties, Dictionary`2 restoreProperties, ILogger[] loggers, LoggerVerbosity verbosity, DistributedLoggerRecord[] distributedLoggerRecords, Int32 cpuCount, Boolean enableNodeReuse, TextWriter preprocessWriter, TextWriter targetsWriter, Boolean detailedSummary, ISet`1 warningsAsErrors, ISet`1 warningsAsMessages, Boolean enableRestore, ProfilerLogger profilerLogger, Boolean enableProfiler, Boolean interactive, Boolean isolateProjects, GraphBuildOptions graphBuildOptions, Boolean lowPriority, String[] inputResultsCaches, String outputResultsCache)

MSBUILD : error MSB1025: An internal failure occurred while running MSBuild.
Microsoft.Build.Shared.InternalErrorException: MSB0001: Internal MSBuild Error: Request 5 expected to be in state Executing but state is actually Yielding
   at Microsoft.Build.CommandLine.MSBuildApp.BuildProject(String projectFile, String[] targets, String toolsVersion, Dictionary`2 globalProperties, Dictionary`2 restoreProperties, ILogger[] loggers, LoggerVerbosity verbosity, DistributedLoggerRecord[] distributedLoggerRecords, Int32 cpuCount, Boolean enableNodeReuse, TextWriter preprocessWriter, TextWriter targetsWriter, Boolean detailedSummary, ISet`1 warningsAsErrors, ISet`1 warningsAsMessages, Boolean enableRestore, ProfilerLogger profilerLogger, Boolean enableProfiler, Boolean interactive, Boolean isolateProjects, GraphBuildOptions graphBuildOptions, Boolean lowPriority, String[] inputResultsCaches, String outputResultsCache)
   at Microsoft.Build.CommandLine.MSBuildApp.Execute(String[] commandLine)
Unhandled exception: Microsoft.Build.Shared.InternalErrorException: MSB0001: Internal MSBuild Error: Request 5 expected to be in state Executing but state is actually Yielding
   at Microsoft.Build.CommandLine.MSBuildApp.BuildProject(String projectFile, String[] targets, String toolsVersion, Dictionary`2 globalProperties, Dictionary`2 restoreProperties, ILogger[] loggers, LoggerVerbosity verbosity, DistributedLoggerRecord[] distributedLoggerRecords, Int32 cpuCount, Boolean enableNodeReuse, TextWriter preprocessWriter, TextWriter targetsWriter, Boolean detailedSummary, ISet`1 warningsAsErrors, ISet`1 warningsAsMessages, Boolean enableRestore, ProfilerLogger profilerLogger, Boolean enableProfiler, Boolean interactive, Boolean isolateProjects, GraphBuildOptions graphBuildOptions, Boolean lowPriority, String[] inputResultsCaches, String outputResultsCache)
   at Microsoft.Build.CommandLine.MSBuildApp.Execute(String[] commandLine)
   at Microsoft.Build.CommandLine.MSBuildApp.Main(String[] args)
   at Microsoft.DotNet.Cli.Utils.MSBuildForwardingAppWithoutLogging.ExecuteInProc(String[] arguments)
```

It looks to be failing on line [403 of the IkvmCompilerLauncher](https://github.com/ikvm-revived/ikvm/blob/develop/src/IKVM.Tool/Compiler/IkvmCompilerLauncher.cs#L403) when trying to load Mono.Unix. I've verified Mono is installed in both the environments I tested with however. Anyone else attempted building on a Linux host yet and had any success?

--- a/src/IKVM.MSBuild.Tasks/IkvmCompiler.cs
+++ b/src/IKVM.MSBuild.Tasks/IkvmCompiler.cs
@@ -1,6 +1,7 @@
 ﻿using System;
 using System.IO;
 using System.Linq;
+using System.Runtime.InteropServices;
 using System.Threading;
 
 using IKVM.Tool.Compiler;
@@ -16,6 +17,20 @@ namespace IKVM.MSBuild.Tasks
     /// </summary>
     public class IkvmCompiler : Task
     {
+
+#if NETCOREAPP
+
+        /// <summary>
+        /// Initializes the static instance.
+        /// </summary>
+        static IkvmCompiler()
+        {
+            // preload Mono.Unix native library, MSBuild isn't capable of following dependency context
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) && Environment.Is64BitProcess)
+                NativeLibrary.Load(Path.Combine(Path.GetDirectoryName(typeof(IkvmCompiler).Assembly.Location), "runtimes", "linux-x64", "native", "libMono.Unix.so"));
+        }
+
+#endif
 
         /// <summary>
         /// Root of the tools director.

